vendor/bat-native-ledger/src/bat/ledger/internal/common/brotli_helpers.cc

```diff
@@ -0,0 +1,132 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ledger/internal/common/brotli_helpers.h"
+
+#include <vector>
+
+#include "third_party/brotli/include/brotli/decode.h"
+
+namespace {
+
+class BrotliStreamDecoder {
```

Comment:
there is already existing code for handling Brotli in chromium BrotliSourceStream

Reply:
in fact most of this code appears to be copied from chromium and there's no reason to do that. We can just use it directly

Reply:
For future reference, reusing `BrotliSourceStream` from `net/filter` would require creating a `SourceStream` class from a string (example [here](https://cs.chromium.org/chromium/src/chromeos/printing/ppd_line_reader.cc?sq=package:chromium&g=0&l=41)).